src/components/reusable/codeView/codeView.ts

```diff
@@ -0,0 +1,246 @@
+import { html, LitElement } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import Prism from 'prismjs';
+
+import copyIcon from '@carbon/icons/es/copy/20';
+import checkmarkIcon from '@carbon/icons/es/checkmark--outline/20';
+
+import '@kyndryl-design-system/shidoka-foundation/components/button';
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+
+import CodeViewStyles from './codeView.scss';
+import PrismStyles from './prismSyntaxStyles.scss';
+
+/**
+ * `<kyn-code-view>` component to display `<code>` snippets, either inline within HTML content or as standalone single-/multi-line block elements.
+ * @fires on-custom-copy - Emits when the copy button is clicked.
+ * @slot unnamed - inline text slot from story book.
+ * @slot inline-example - Slot for non-code example text to preceed inline code snippet.
+ * @slot copy-slot - Slot for copy button.
+ */
+@customElement('kyn-code-view')
+export class CodeView extends LitElement {
+  static override styles = [CodeViewStyles, PrismStyles];
+
+  /** Code snippet type: `inline`, `block` */
+  @property({ type: String })
+  snippetType = 'block';
+
+  /** For `block` snippetType only -- size: `auto`, `sm`, `md`, or `lg`. */
+  @property({ type: String })
+  size = 'md';
+
+  /** Code language (ex: `javascript`, `css`, `markdown`, `xml`). */
+  @property({ type: String })
+  language = '';
+
+  /** Optional title to be displayed above code snippet. */
+  @property({ type: String })
+  override title = '';
```

Comment:
I'd suggest changing this from `title`, so it doesn't apply the native html "title" attribute/tooltip.

Reply:
i've created properties to be set by the user for `ariaLabelAttr` (inline component), as well as title and description attribute properties (`copyButtonTitleAttr` `copyButtonDescriptionAttr`)

that title/label above the block code snippet has been changed to `codeViewLabel`